File: ProjetoConsulta.Web/ClientApp/src/model/marcacao.ts
import { Data } from "@angular/router";
import { Timestamp } from "rxjs";

export class Marcacao {

  id: string;
  dataMarcacao: Data;
  disponivel: boolean;
  horario: Data;
  pacienteID: string;
  hospitalID: string;

  constructor(_id: string, _dataMarcacao: Data, _disponivel: boolean, _horario: Data, _pacienteID: string, _hospitalID: string) {
    this.id = _id;
    this.dataMarcacao = _dataMarcacao;
    this.disponivel = _disponivel;
    this.horario = _horario;
    this.pacienteID = _pacienteID;
    this.hospitalID = _hospitalID;
 
  }

}

File: ProjetoConsultaPaciente.Dominio/Entidade/Marcacao.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ProjetoConsultaPaciente.Dominio.Entidade
{
    public class Marcacao
    {
        public System.Guid Id { get; set; }
        public System.DateTime DataMarcacao { get; set; }
        public bool Disponivel { get; set; }
        public System.TimeSpan Horario { get; set; }

        public Guid? PacienteId { get; set; }
        public virtual Paciente Paciente { get; set; }

        public Guid HospitalId { get; set; }
        public virtual Hospital Hospital { get; set; }
    }
}
